plugins/modules/zendesk_ticket.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Luis Valle (levalle232@gmail.com)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: zendesk_ticket
+short_description: Manages tickets in Zendesk
+description:
+  - This module allows you to create and close tickets in Zendesk.
+  - Authentication is handled by the ZENDESK_API class.
+author: "Luis Valle (@elchico2007)"
+version_added: 9.4.0
+extends_documentation_fragment:
+    - community.general.attributes
+attributes:
+  check_mode:
+    support: none
+  diff_mode:
+    support: none
+options:
+  url:
+    type: str
+    description:
+      - The URL of the Zendesk instance.
+    required: true
+  username:
+    type: str
+    description:
+      - The Zendesk account username.
+    required: true
+    aliases: ['user']
+  password:
+    type: str
+    description:
+      - The Zendesk account password.
+      - Optional if token is used.
+    required: false
+    aliases: ['pass']
+  token:
+    type: str
+    description:
+      - The API token for authentication.
+      - Optional if password is used.
+    required: false
+  body:
+    type: str
+    description:
+      - The body of the ticket.
+    default: ''
+  priority:
+    type: str
+    description:
+      - The priority of the ticket.
+    choices: ['urgent', 'high', 'normal', 'low']
+    default: normal
+  status:
+    type: str
+    description:
+      - The status of the ticket.
+      - The 'new' choice is not idempotent and will create a new ticket each time it's used.
+      - The 'closed' choice will close the ticket. If a body is provided, it will mark the ticket as resolved with the given resolution.
```

Comment:
```suggestion
      - The V(new) choice is not idempotent and will create a new ticket each time it's used.
      - The V(closed) choice will close the ticket. If O(body) is provided, it will mark the ticket as resolved with the given resolution.
```